lib/common/ipc.c

```diff
@@ -579,20 +579,18 @@ crm_ipc_prepare(uint32_t request, xmlNode * message, struct iovec ** result, uin
 
             free(buffer);
 
-            if (header->size_compressed > biggest) {
-                biggest = 2 * QB_MAX(header->size_compressed, biggest);
-            }
+            biggest = QB_MAX(header->size_compressed, biggest);
 
         } else {
             ssize_t rc = -EMSGSIZE;
 
             crm_log_xml_trace(message, "EMSGSIZE");
-            biggest = 2 * QB_MAX(header->size_uncompressed, biggest);
+            biggest = QB_MAX(header->size_uncompressed, biggest);
```

Comment:
> As long as the following message cannot fit into the current buffer, even if it's only 1 byte larger, it'll warn again anyway, right?

Thats my problem :-)
We don't want the same message 10 times in a few seconds if 10 updates grow the CIB by 1 byte each.

Just suggest a reasonable threshold and don't warn until that too would have been exhausted.